core/Container/StaticContainer.php

```diff
@@ -75,8 +75,9 @@ public static function addDefinitions(array $definitions)
     /**
      * Proxy to Container::get()
      *
-     * @param string $name Container entry name.
-     * @return mixed
+     * @template T of object
+     * @param class-string<T>|string $name Container entry name.
+     * @return ($name is class-string<T> ? T : mixed)
```

Comment:
This should ensure that for something like `StaticContainer('Piwik/Auth')` an object of `Piwik/Auth` will be expected as returned value.